### DISCUSSION TITLE
Hoverscroll over Dropdown boxes

### DISCUSSION THREAD

**Discussion by sebastian-s-beckmann** [obsproject/obs-studio#4980] [General]
At the moment, OBS dropdown lists, everywhere in the app, have the behaviour that if the hovering mouse is over them, scrolling with the mouse wheel will change the current selection (I'm not sure what the technical term is, I'll call it "hoverscroll" for now).

There has been a lively discussion (on the discord server) about whether this is behaviour that is useful and intentional. The result has been to move it to a place where more people may look and information doesn't get lost as easily, therefore this GH discussion is created.
The particular problem was about the source toolbar, but this applies to most of the dropdown lists.

Here are the major discussion points made so far:
- In OBS, many settings immediately come into effect once selected. This also applies to dropdown lists. Here, it is rather easy to accidentally change the selected item. Immediately taking effect, this could screw up which display you're showing in display capture, which game you're streaming in game capture, change OBS's language in the settings, etc. Having hoverscroll removed would significantly lower the risk of things like this happening (especially in the toolbar).
- There perhaps aren't many people using hoverscroll, or even knowing about it. Removing it probably won't hurt that much.
- We don't know how many people actually use it, it might be important for some peoples workflows.

There were three solutions suggested:
- Leave it as is. This would keep the problems above.
- Only disable it for the source toolbar. This would be bad for continuity across the app.
- Disable it everywhere. Again, see above.

Basically, the question is whether it does more harm than good.

Any insight and more opinions on this, especially from people who use hoverscroll, would be appreciated.

**Comment by MattyFresh68** [comment]
Upon going through the various combo boxes myself, I do not think there is a single one I personally use with scroll. I think disabling scroll in these globally doesn't hurt user experience. Though additional feedback would be nice. Without a broad base using scroll, I think the simple "click to expand options" is suitable.

**Comment by Fenrirthviti** [comment]
Personally, I would prefer to allow scrolling to change options only when they are actively selected and in focus (i.e. they were clicked on first). If they are not in focus, they should not be able to be changed by scrolling.

**Reply by Fenrirthviti** [reply]
To this point, it might be worth checking to see if it's possible to ensure all options are "unfocused" by default when a UI element is opened.

**Comment by WizardCM** [comment]
We previously did the same for sliders in #1811 and #1837 - I would be OK with a change like this, especially in the Source Toolbar.
